lib/smart_answer/money.rb

```diff
@@ -32,8 +25,21 @@ def <=>(other)
       end
     end
 
-    def self.valid?(raw_input)
-      raw_input.is_a?(Numeric) || raw_input.is_a?(Money) || raw_input =~ /\A *[0-9]+(\.[0-9]{1,2})? *\z/
+    def self.raw_input_to_number(raw_input)
+      raw_input = raw_input.is_a?(Numeric) ? raw_input : raw_input.to_s.delete(',').gsub(/\s/, '')
+      if raw_input.to_f.infinite?
+        raise InvalidResponse, "Sorry, that number is too big. Please try again.", caller
+      else
+        valid_raw_input(raw_input)
+      end
+    end
+
+    def self.valid_raw_input(raw_input)
```

Comment:
Would be nice to keep the validation predicate separate from the validation method. So, I'd expect this to be two methods:

- `valid_raw_input?`, which returns a boolean
- `validate_raw_input`, which calls `valid_raw_input?` and will `raise` if it's invalid. You could also move the `infinite?` check into the validation, because it's part of what makes the input valid.